lib/src/testing/utils/testrunner.dart

```diff
@@ -746,6 +765,23 @@ void _runTests<PrivateKey, PublicKey>(
       publicKey = pair.publicKey;
       privateKey = pair.privateKey;
     });
+  } else if (c.importKeyException != null) {
+    test('pkcs8 import exception', () async {
+      try {
+      await r._importPrivatePkcs8Key!(c.privatePkcs8KeyData!, {'curve': 'p-521'});
+      check(false, 'Expected an exception for P-512 import');
+    } catch (e) {
+      check(e.toString().contains(c.pkcs8ImportKeyException!));
+    }
+    });
+    test('jwk import exception', () async {
+      try {
+      await r._importPrivateJsonWebKey!(c.privateJsonWebKeyData!, {'curve': 'p-521'});
+      check(false, 'Expected an exception for P-512 import');
+    } catch (e) {
+      check(e.toString().contains(c.jwkImportKeyException!));
+    }
+    });
```

Comment:
If `importKeyException` is not `null`, then we can test that importing public and private keys fail, but all the other tests we can't do, right?

So maybe it's fine to just do a `return;` here? That way we don't have to add `&& c.pkcs8ImportKeyException == null` to all the other test cases.

Reply:
We should also when doing this, make sure that in `_validateTestCase` we validate that the test case doesn't have values that won't be used.

In particular we need to validate that:
 * if `generateKeyParams` is not null, then `importKeyException` must be `null`!
 * if `importKeyException` is not null, then
   * All the other properties that won't be used must be `null`.
     * This includes `plaintext`, `signature`, `ciphertext`, `derivedBits`, `derivedLength` and more.
       (If the tests don't use the values, they MUST not be specified, otherwise the test case will be hard to read and understand)
     * Probably also includes `signVerifyParams`, etc. since again, we can't use them for anything, so they shouldn't be specified.
   * At-least one of:
     * `publicRawKeyData`, `publicSpkiKeyData`, `publicJsonWebKeyData`, `privateRawKeyData`, `privatePkcs8KeyData`, or `privateJsonWebKeyData` must be specified.
     * Note: that we probably don't need all of them to be specified.
     * Note: it's probably okay if there is no private key or no public key, just so long as there is at-least ONE of these properties provided. Normally, when `importKeyException == null`, this is ofcourse not allowed.

Reply:
Added a `return;` in this section. Removed recurring `&& c.pkcs8ImportKeyException == null`
Updated `_validateTestCase` with the above checks